tests/Tests.AzureAppConfiguration/Integration/IntegrationTests.cs

```diff
@@ -0,0 +1,544 @@
+                .AddAzureAppConfiguration(options =>
+                {
+                    options.Connect(GetConnectionString());
+                    options.Select($"{TestKeyPrefix}:*");
+                })
+                .Build();
+
+            // Assert
+            Assert.Equal("InitialValue1", config[$"{TestKeyPrefix}:Setting1"]);
+            Assert.Equal("InitialValue2", config[$"{TestKeyPrefix}:Setting2"]);
+        }
+
+        [Fact]
+        public async Task TryRefreshAsync_UpdatesConfiguration_WhenSentinelKeyChanged()
+        {
+            Skip.If(_skipTests, _skipReason);
+
+            // Arrange - Setup test-specific keys
+            var (keyPrefix, sentinelKey, _) = await SetupTestKeys("UpdatesConfig");
+            IConfigurationRefresher refresher = null;
+
+            var config = new ConfigurationBuilder()
+                .AddAzureAppConfiguration(options =>
+                {
+                    options.Connect(GetConnectionString());
+                    options.Select($"{keyPrefix}:*");
+                    options.ConfigureRefresh(refresh =>
+                    {
+                        refresh.Register(sentinelKey, refreshAll: true)
+                              .SetRefreshInterval(TimeSpan.FromSeconds(1));
+                    });
+
+                    refresher = options.GetRefresher();
+                })
+                .Build();
+
+            // Verify initial values
+            Assert.Equal("InitialValue1", config[$"{keyPrefix}:Setting1"]);
+
+            // Update values in the store
+            await _configClient.SetConfigurationSettingAsync(new ConfigurationSetting($"{keyPrefix}:Setting1", "UpdatedValue1"));
+            await _configClient.SetConfigurationSettingAsync(new ConfigurationSetting(sentinelKey, "Updated"));
+
+            // Wait for cache to expire
+            await Task.Delay(TimeSpan.FromSeconds(2));
+
+            // Act
+            var result = await refresher.TryRefreshAsync();
+
+            // Assert
+            Assert.True(result);
+            Assert.Equal("UpdatedValue1", config[$"{keyPrefix}:Setting1"]);
+        }
+
+        [Fact]
+        public async Task TryRefreshAsync_RefreshesOnlySelectedKeys_WhenUsingKeyFilter()
+        {
+            Skip.If(_skipTests, _skipReason);
+
+            // Arrange - Setup test-specific keys
+            var (keyPrefix, sentinelKey, _) = await SetupTestKeys("RefreshesSelectedKeys");
+            IConfigurationRefresher refresher = null;
+
+            var config = new ConfigurationBuilder()
+                .AddAzureAppConfiguration(options =>
+                {
+                    options.Connect(GetConnectionString());
+                    options.Select($"{keyPrefix}:*");
+
+                    // Only refresh Setting1 when sentinel changes
+                    options.ConfigureRefresh(refresh =>
+                    {
+                        refresh.Register(sentinelKey, $"{keyPrefix}:Setting1", refreshAll: false)
```

Comment:
Is this registering a key with a label `$"{keyPrefix}:Setting1"` ? That looks wrong.

Reply:
Yea I'll take a closer look at the unit tests now, posted draft a bit early